lxd/storage_dir.go

```diff
@@ -58,14 +59,11 @@ func (s *storageDir) StoragePoolCreate() error {
 		source = filepath.Join(shared.VarPath("storage-pools"), s.pool.Name)
 		s.pool.Config["source"] = source
 	} else {
-		cleanSource := filepath.Clean(source)
-		lxdDir := shared.VarPath()
-		poolMntPoint := getStoragePoolMountPoint(s.pool.Name)
-		if strings.HasPrefix(cleanSource, lxdDir) && cleanSource != poolMntPoint {
-			return fmt.Errorf("DIR storage pool requests in LXD directory \"%s\" are only valid under \"%s\"\n(e.g. source=%s)", shared.VarPath(), shared.VarPath("storage-pools"), poolMntPoint)
-		}
```

Comment:
Why is this check being removed? Preventing the user from bind-mounting a path that's under /var/lib/lxd seems like a good idea and should be kept (and made consistent with btrfs if not done already).